Packs/SplunkPy/Integrations/SplunkPyV2/SplunkPyV2.py

```diff
@@ -0,0 +1,4180 @@
+
+
+# =========== Time & Date Utilities ===========
+
+
+def get_current_splunk_time(splunk_service: client.Service):
+    get_time_query = f'| gentimes start=-1 | eval clock = strftime(time(), "{ISO_FORMAT_TZ_AWARE}") | sort 1 -_time | table clock'
+    search_results = splunk_service.jobs.oneshot(get_time_query, count=1, output_mode=OUTPUT_MODE_JSON)
+
+    reader = results.JSONResultsReader(search_results)
+    for item in reader:
+        if isinstance(item, dict):
+            return item["clock"]
+        if handle_message(item):
+            continue
+
+    raise ValueError("Error: Could not fetch Splunk time")
+
+
+def extract_timezone_offset_from_splunk_time(splunk_time_str: str) -> str:
+    """Extract timezone offset from Splunk time string.
+
+    Args:
+        splunk_time_str: Time string in ISO_FORMAT_TZ_AWARE format
+                        (e.g., '2025-12-03T11:53:45.138540+02:00')
+
+    Returns:
+        Timezone offset string (e.g., '+02:00', '-05:00')
+        Returns '+00:00' if extraction fails
+    """
+    try:
+        # Parse the datetime string to extract timezone
+        dt = datetime.strptime(splunk_time_str, ISO_FORMAT_TZ_AWARE)
+        # Get the timezone offset
+        tz_offset = dt.strftime("%z")
+        # Format as +HH:MM
+        return f"{tz_offset[:3]}:{tz_offset[3:]}"
+    except Exception as e:
+        demisto.error(f"Failed to extract timezone from '{splunk_time_str}': {e}")
+        return "+00:00"  # Default to UTC
+
+
+def get_splunk_timezone_offset(service: client.Service) -> str:
+    """Get Splunk server timezone offset with caching.
+
+    Retrieves the timezone offset from integration context cache.
+    If not cached, queries Splunk server and caches the result.
+
+    Args:
+        service: Splunk service object
+
+    Returns:
+        Timezone offset string (e.g., '+02:00')
+    """
+    TIMEZONE_CACHE_KEY = "splunk_timezone_offset"
+
+    # Try to get from cache
+    integration_context = get_integration_context()
+    cached_timezone = integration_context.get(TIMEZONE_CACHE_KEY)
+
+    if cached_timezone:
+        demisto.debug(f"Using cached Splunk timezone: {cached_timezone}")
+        return cached_timezone
+
+    # Not cached - query Splunk
+    demisto.debug("Timezone not cached, querying Splunk server")
+    try:
+        splunk_time = get_current_splunk_time(service)
+        timezone_offset = extract_timezone_offset_from_splunk_time(splunk_time)
+
+        # Cache the timezone
+        integration_context[TIMEZONE_CACHE_KEY] = timezone_offset
+        set_integration_context(integration_context)
+
+        demisto.debug(f"Cached Splunk timezone: {timezone_offset}")
+        return timezone_offset
+
+    except Exception as e:
+        demisto.error(f"Failed to get Splunk timezone: {e}")
+        return "+00:00"  # Default to UTC on error
+
+
+def enforce_look_behind_time(fetch_window_start_time, fetch_window_end_time, look_behind_time):
```

Comment:
can you please change the phrase to `lookback`? this will better match our name conventions.